### PR TITLE
make export use filters, add tests

### PR DESCRIPTION
**Pull request by jacalata**
1. updated export command to look for filter params on the url given and convert them to RequestOption view_filters
2. fixed bytes v iterator for data files.
3. added tests

### CODE REVIEW

File: tests/commands/test_geturl_utils.py [hunk 1]
```diff
@@ -59,9 +67,70 @@ def test_view_name(self):
...
+        wb_url = "wb-name/view-name/kitty"
+        view, wb = ExportCommand.parse_export_url_to_workbook_and_view(mock_logger, wb_url)
+        assert view is None
+        assert wb is None
+
+    def test_extract_query_params(self):
+        url = "wb-name/view-name?param1=value1"
+        options = TSC.PDFRequestOptions()
+        updated_options = ExportCommand.extract_query_params(options, url)
+        assert updated_options is not None
```
Comment (andrew-young-tableau) [inline tests/commands/test_geturl_utils.py:94]: Validate `updated_options.vf` got updated too?

File: tabcmd/commands/datasources_and_workbooks/export_command.py [hunk 2]
```diff
@@ -73,65 +56,103 @@ def run_command(args):
...
+    def download_png(server, view_item, url, logger):
+        logger.trace(url)
         req_option_image = TSC.ImageRequestOptions(maxage=1)
+        req_option_image = ExportCommand.extract_query_params(req_option_image, url)
         server.views.populate_image(view_item, req_option_image)
         return view_item.png
 
     @staticmethod
     def parse_export_url_to_workbook_and_view(logger, url):
+        filters = ""
```
Comment (andrew-young-tableau) [inline tabcmd/commands/datasources_and_workbooks/export_command.py:150]: Unused?

File: tabcmd/commands/datasources_and_workbooks/export_command.py [hunk 3]
```diff
@@ -73,65 +56,103 @@ def run_command(args):
...
+                return None
+            params = query.split("&")
+            logger.trace(params)
+            for value in params:
+                filter = value.split("=")
+                request_options.vf(filter[0], filter[1])
+        except BaseException as e:
+            logger.error("Error building filter params", e)
+            ExportCommand.log_stack(logger)  # type: ignore
+        return request_options
```
Comment (andrew-young-tableau) [inline tabcmd/commands/datasources_and_workbooks/export_command.py:113]: Do we even need to return `request_options` here? I'm assuming the original `request_options` argument passed in will have its `view_filters` field updated.

File: tabcmd/commands/datasources_and_workbooks/export_command.py [hunk 4]
```diff
@@ -73,65 +56,103 @@ def run_command(args):
         if not view_content_url and not wb_content_url:
             Errors.exit_with_error(logger, _("export.errors.requires_workbook_view_param").format(ExportCommand))
 
-        try:
+        print(args.pagelayout, args.pagesize, args.filename, args.width, args.height, args.filter)
```
[the remark this answers is not in the document]
Comment (jacalata) [reply tabcmd/commands/datasources_and_workbooks/export_command.py]: fixed, changed to logger.debug

File: tabcmd/commands/datasources_and_workbooks/export_command.py [hunk 5]
```diff
@@ -73,65 +56,103 @@ def run_command(args):
...
+                ExportCommand.save_to_data_file(logger, output, save_name)
+            else:
+                ExportCommand.save_to_file(logger, output, save_name)
 
         except Exception as e:
             Errors.exit_with_error(logger, "Error saving to file", e)
 
     @staticmethod
-    def download_wb_pdf(server, workbook_item):
+    def extract_query_params(request_options: TSC.PDFRequestOptions, url, logger=None):
```
[the remark this answers is not in the document]
Comment (jacalata) [reply tabcmd/commands/datasources_and_workbooks/export_command.py]: renamed the method to be clearer about what it does

File: tabcmd/commands/datasources_and_workbooks/export_command.py (hunk 3 above)
Comment (jacalata) [reply tabcmd/commands/datasources_and_workbooks/export_command.py:113]: good call, we do not. Fixed.

Comment (anyoung-tableau) [reply tabcmd/commands/datasources_and_workbooks/export_command.py:113]: Was this one still pending?

**Pull request merged**